### PR TITLE
feat(mobile): ai chat service bb-441

### PR DESCRIPTION
**Pull request by KotenkoNata**
-  Display categories with 3 lowest values.
- Create a service to handle backend chat.
 - Add a "Send Request" button (needs to be refined).
 - Send requests with selected categories and display the tasks that have been received.

Preview 

[3+lowest.webm](https://github.com/user-attachments/assets/dbda987d-bd28-4666-a2b8-220c66c2ba0d)

[checkbox.webm](https://github.com/user-attachments/assets/c54675d8-2268-4c0e-9099-d62bbcc0be6c)

### CODE REVIEW

File: apps/mobile/src/screens/chat/chat.tsx [hunk 1]
```diff
@@ -1,72 +1,96 @@
...
 
+	const handleKeyExtractor = useCallback(
+		(item: ChatMessageItem, index: number) => index.toString(),
+		[],
+	);
+
 	return (
-		<ScreenWrapper style={[globalStyles.flex1, styles.container]}>
-			<ScrollView
+		<ScreenWrapper edges={[]} style={[globalStyles.flex1, styles.container]}>
```
Comment (fshabanov) [inline apps/mobile/src/screens/chat/chat.tsx:57]:
```suggestion
		<ScreenWrapper style={[globalStyles.flex1, styles.container]}>
```
Can we remove it, if we don't pass anything?

File: apps/mobile/src/screens/chat/libs/components/chat-box/libs/components/components.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+export { AiTask } from "./ai-task/ai-task";
```
Comment (fshabanov) [inline apps/mobile/src/screens/chat/libs/components/chat-box/libs/components/components.ts:1]:
```suggestion
export { AITask } from "./ai-task/ai-task";
```

File: apps/mobile/src/screens/chat/libs/components/chat-buttons/libs/components/checkbox-buttons/checkbox-buttons.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+					author: ChatMessageAuthor.USER,
+					text: "Confirm tasks generation",
+				}),
+			);
+		},
+		[dispatch, quizCategories, threadId],
+	);
+
+	return (
+		<ChatMessage isUser={false} text={CHECKBOX_SELECTOR_TEXT}>
```
Comment (fshabanov) [inline apps/mobile/src/screens/chat/libs/components/chat-buttons/libs/components/checkbox-buttons/checkbox-buttons.tsx:55]:
```suggestion
		<ChatMessage text={CHECKBOX_SELECTOR_TEXT}>
```
Let's make false by default

File: apps/mobile/src/screens/chat/libs/components/chat-buttons/libs/components/generate-task-buttons/generate-task-buttons.tsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+	const threadId = useAppSelector((state) => state.chat.threadId);
+
+	const lowestCategories = getLowestCategories(scores);
+
+	const handleNo = useCallback(() => {
+		dispatch(chatActions.setButtonsMode(ButtonsMode.CHECKBOX));
+		dispatch(
+			chatActions.addTextMessage({
+				author: ChatMessageAuthor.USER,
+				text: "No smth else",
```
Comment (fshabanov) [inline apps/mobile/src/screens/chat/libs/components/chat-buttons/libs/components/generate-task-buttons/generate-task-buttons.tsx:29]:
```suggestion
				text: "No, something else",
```

File: apps/mobile/src/screens/chat/libs/components/chat-buttons/libs/components/generate-task-buttons/generate-task-buttons.tsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+				text: "Yes, 3 lowest",
+			}),
+		);
+	}, [dispatch, threadId, lowestCategories]);
+
+	return (
+		<ChatMessage isUser={false} text={CONFIRMATION_TEXT}>
+			<View style={[globalStyles.gap12, globalStyles.mt12]}>
+				<Button label="Yes, 3 lowest" onPress={handleYes} />
+				<Button appearance="outlined" label="No smth else" onPress={handleNo} />
```
Comment (fshabanov) [inline apps/mobile/src/screens/chat/libs/components/chat-buttons/libs/components/generate-task-buttons/generate-task-buttons.tsx:63]:
```suggestion
				<Button appearance="outlined" label="No, something else" onPress={handleNo} />
```

File: apps/mobile/src/screens/chat/libs/components/chat-buttons/libs/components/generate-task-buttons/generate-task-buttons.tsx [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+		dispatch(chatActions.setButtonsMode(ButtonsMode.CHECKBOX));
+		dispatch(
+			chatActions.addTextMessage({
+				author: ChatMessageAuthor.USER,
+				text: "No smth else",
+			}),
+		);
+	}, [dispatch]);
+
+	const handleYes = useCallback(() => {
```
Comment (fshabanov) [inline apps/mobile/src/screens/chat/libs/components/chat-buttons/libs/components/generate-task-buttons/generate-task-buttons.tsx:34]: Not a clear naming

File: apps/mobile/src/screens/chat/libs/components/loader/libs/enums/delay.enum.ts [hunk 7]
```diff
@@ -0,0 +1,7 @@
+const DotDelays = {
```
Comment (fshabanov) [inline apps/mobile/src/screens/chat/libs/components/loader/libs/enums/delay.enum.ts:1]:
```suggestion
const DotDelay = {
```
Change file name to `dot-delay.enum.ts`

File: apps/mobile/src/screens/chat/libs/components/loader/libs/helpers/get-animated-styles.helper.ts [hunk 8]
```diff
@@ -0,0 +1,12 @@
+import { type SharedValue, useAnimatedStyle } from "react-native-reanimated";
+
+type Property = {
+	transform: { scale: number }[];
+};
+
+const getAnimatedStyle = (scaleValue: SharedValue<number>): Property =>
```
Comment (fshabanov) [inline apps/mobile/src/screens/chat/libs/components/loader/libs/helpers/get-animated-styles.helper.ts:7]: Let's convert it to a custom hook. Hooks cannot be called inside of helpers
image

File: apps/mobile/src/libs/components/chat-header/styles.ts [hunk 9]
```diff
@@ -0,0 +1,15 @@
+import { StyleSheet } from "react-native";
+
+import { BaseColor } from "~/libs/enums/enums";
+
+const styles = StyleSheet.create({
+	header: {
+		backgroundColor: BaseColor.BG_WHITE,
+		borderBottomColor: BaseColor.EXTRA_LIGHT_GRAY,
+		borderBottomWidth: 1,
+		display: "flex",
```
Comment (revenets) [inline apps/mobile/src/libs/components/chat-header/styles.ts:10]:
```suggestion
```

File: apps/mobile/src/screens/chat/libs/components/loader/loader.tsx [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+import { globalStyles } from "~/libs/styles/styles";
+
+import { ANIMATION_CONFIG } from "./libs/constants/constants";
+import { DotDelays } from "./libs/enums/enums";
+import { getAnimatedStyle } from "./libs/helpers/helpers";
+import { styles } from "./styles";
+
+const INITIAL_SCALE = -1;
+const TARGET_SCALE = 1.5;
+const INITIAL_SCALE_VALUE = 1;
```
Comment (fshabanov) [inline apps/mobile/src/screens/chat/libs/components/loader/loader.tsx:20]: Create enum

File: apps/mobile/src/slices/chat/actions.ts [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+	type ThreadMessageCreateDto,
+} from "~/packages/chat/chat";
+
+import { name as sliceName } from "./chat.slice";
+
+const initConversation = createAsyncThunk<
+	ThreadMessageCreateDto,
+	undefined,
+	AsyncThunkConfig
+>(`${sliceName}/init-conversation`, async (_, { extra }) => {
```
Comment (fshabanov) [inline apps/mobile/src/slices/chat/actions.ts:23]:
```suggestion
>(`${sliceName}/initialize-conversation`, async (_, { extra }) => {
```

File: apps/mobile/src/libs/components/chat-message/styles.ts [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+import { BaseColor } from "~/libs/enums/enums";
+
+const styles = StyleSheet.create({
+	container: {
+		backgroundColor: BaseColor.BG_LIGHT,
+		borderBottomLeftRadius: 10,
+		borderBottomRightRadius: 10,
+		borderTopLeftRadius: 0,
+		borderTopRightRadius: 10,
+		flex: 1,
```
Comment (revenets) [inline apps/mobile/src/libs/components/chat-message/styles.ts:12]: Use globalStyles instead
```suggestion
```

File: apps/mobile/src/screens/chat/chat.tsx [hunk 13]
```diff
@@ -1,72 +1,96 @@
...
+							styles.screenWrapper,
+						]}
+						data={messages}
+						inverted
+						keyExtractor={handleKeyExtractor}
+						ListFooterComponent={
+							<>
+								{threadId && !isLoading && <ChatButtons />}
+								{isLoading && <Loader />}
+							</>
```
Comment (revenets) [inline apps/mobile/src/screens/chat/chat.tsx:87]:
```suggestion
							<>
								{isLoading ? <Loader /> : (
								   threadId && <ChatButtons />
								)}
							</>
```

File: apps/mobile/src/screens/chat/libs/components/chat-buttons/libs/components/generate-task-buttons/generate-task-buttons.tsx [hunk 14]
```diff
@@ -0,0 +1,69 @@
...
+
+const GenerateTaskButtons: React.FC = () => {
+	const dispatch = useAppDispatch();
+
+	const scores = useAppSelector((state) => state.quiz.scores);
+	const threadId = useAppSelector((state) => state.chat.threadId);
+
+	const lowestCategories = getLowestCategories(scores);
+
+	const handleNo = useCallback(() => {
```
Comment (revenets) [inline apps/mobile/src/screens/chat/libs/components/chat-buttons/libs/components/generate-task-buttons/generate-task-buttons.tsx:24]:
```suggestion
	const handleNoPress = useCallback(() => {
```

File: apps/mobile/src/screens/chat/libs/components/chat-buttons/libs/components/generate-task-buttons/generate-task-buttons.tsx (hunk 6 above)
Comment (revenets) [inline apps/mobile/src/screens/chat/libs/components/chat-buttons/libs/components/generate-task-buttons/generate-task-buttons.tsx:34]:
```suggestion
	const handleYesPress = useCallback(() => {
```

**Review by revenets — CHANGES_REQUESTED** [review]

File: apps/mobile/src/navigations/bottom-tabs-navigator/bottom-tabs-navigator.tsx [hunk 15]
```diff
@@ -39,9 +43,14 @@ const BottomTabsNavigator: React.FC = () => {
 				component={Chat}
 				name={BottomTabScreenName.CHAT}
 				options={{
+					header: (): React.ReactNode => {
+						return <ChatHeader />;
+					},
```
[the remark this answers is not in the document]
Comment (KotenkoNata) [reply apps/mobile/src/navigations/bottom-tabs-navigator/bottom-tabs-navigator.tsx:48]: No, doesn't work.

File: apps/mobile/src/navigations/bottom-tabs-navigator/bottom-tabs-navigator.tsx [hunk 16]
```diff
@@ -39,9 +43,14 @@ const BottomTabsNavigator: React.FC = () => {
...
+					headerShown: true,
```
[the remark this answers is not in the document]
Comment (KotenkoNata) [reply apps/mobile/src/navigations/bottom-tabs-navigator/bottom-tabs-navigator.tsx:49]: Yes, but for other screens, it is false. That's why it is true.

**Comment by KotenkoNata** [comment]
Add regenerate tasks. 

[regenerate.webm](https://github.com/user-attachments/assets/1210fbbf-b746-4117-8698-c94ba44dada6)

File: apps/mobile/src/navigations/bottom-tabs-navigator/bottom-tabs-navigator.tsx (hunk 16 above)
[the remark this answers is not in the document]
Comment (fshabanov) [reply apps/mobile/src/navigations/bottom-tabs-navigator/bottom-tabs-navigator.tsx:49]: Not sure to understand. If it is `true` by default, then we don't need to add this

File: apps/mobile/src/libs/components/chat-message/chat-message.tsx [hunk 17]
```diff
@@ -0,0 +1,62 @@
...
+	style?: StyleProp<ViewStyle>;
+	text?: string;
+};
+
+const ICON_SIZE = 34;
+
+const ChatMessage: React.FC<Properties> = ({
+	children,
+	content,
+	isUser,
```
Comment (fshabanov) [inline apps/mobile/src/libs/components/chat-message/chat-message.tsx:23]:
```suggestion
	isUser = false,
```

File: apps/mobile/src/packages/categories/categories.ts [hunk 18]
```diff
@@ -1 +1 @@
-export { categoriesSelectedValidationSchema } from "./libs/validation-schemas/validation-schemas";
+export { categoriesSelectedValidationSchema } from "./libs/validation-schemas";
```
Comment (fshabanov) [inline apps/mobile/src/packages/categories/categories.ts:1]:
```suggestion
export { categoriesSelectedValidationSchema } from "./libs/validation-schemas/validation-schemas";
```

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/mobile/src/navigations/bottom-tabs-navigator/bottom-tabs-navigator.tsx (hunk 16 above)
[the remark this answers is not in the document]
Comment (KotenkoNata) [reply apps/mobile/src/navigations/bottom-tabs-navigator/bottom-tabs-navigator.tsx:49]: When I create Root navigation, I turn headerShown to false for all NavigationItem. That's why for AI chat I need  `headerShown: true`

**Comment by revenets** [comment]
> Add regenerate tasks.
> 
>  regenerate.webm

I think it would be great to add some extra padding at the bottom, because when the chat is full screen, the last message appears to close to the native navigation tab

**Comment by KotenkoNata** [comment]
Extra padding at the bottom

it was before 
Screenshot 2024-09-25 at 15 53 48

add more space
Screenshot 2024-09-25 at 15 53 24

**Comment by KotenkoNata** [comment]
Do we need to have the exact text and buttons with web version? I have a look to [PR](https://github.com/BinaryStudioAcademy/bsa-2024-bebalance/pull/576#issue-2548091067) and we have differences.

**Comment by revenets** [comment]
Please fix unresolved comments

File: apps/mobile/src/libs/components/chat-message/chat-message.tsx [hunk 19]
```diff
@@ -0,0 +1,62 @@
...
+	children,
+	content,
+	isUser = false,
+	style,
+	text = "",
+}: Properties) => {
+	const messageText = content?.text ?? text;
+
+	const Avatar = isUser ? (
+		<Icon color={BaseColor.LIGHT_GRAY} name="account-circle" size={ICON_SIZE} />
```
Comment (fshabanov) [inline apps/mobile/src/libs/components/chat-message/chat-message.tsx:30]: Let's use user avatar here

File: apps/mobile/src/navigations/bottom-tabs-navigator/bottom-tabs-navigator.tsx [hunk 20]
```diff
@@ -39,9 +43,14 @@ const BottomTabsNavigator: React.FC = () => {
...
 					tabBarIcon: ({ focused }) => (
 						<GradientTabIcon isFocused={focused} name="sms" />
 					),
+					tabBarStyle: { display: "none" },
```
[the remark this answers is not in the document]
Comment (fshabanov) [reply apps/mobile/src/navigations/bottom-tabs-navigator/bottom-tabs-navigator.tsx:53]: Not resolved

File: apps/mobile/src/packages/chat/chat-api.ts [hunk 21]
```diff
@@ -0,0 +1,64 @@
...
+				hasAuth: true,
+				method: "POST",
+				payload: JSON.stringify(payload),
+			},
+		);
+
+		return await response.json<AIAssistantResponseDto>();
+	}
+
+	public async initiateConversation(): Promise<ThreadMessageCreateDto> {
```
[the remark this answers is not in the document]
Comment (fshabanov) [reply apps/mobile/src/packages/chat/chat-api.ts:66]: Not resolved

File: apps/mobile/src/packages/chat/libs/helpers/check-is-task-type.helper.ts [hunk 22]
```diff
@@ -0,0 +1,10 @@
+import { ChatMessageType } from "../enums/enums";
+import { type ChatMessageDto, type TaskMessage } from "../types/types";
+
+const checkIsTaskType = (
+	message: ChatMessageDto,
+): message is ChatMessageDto<TaskMessage> => {
+	return message.type === ChatMessageType.TASK;
+};
```
[the remark this answers is not in the document]
Comment (fshabanov) [reply apps/mobile/src/packages/chat/libs/helpers/check-is-task-type.helper.ts:8]: Not resolved

File: apps/mobile/src/screens/chat/chat.tsx (hunk 1 above)
Comment (fshabanov) [reply apps/mobile/src/screens/chat/chat.tsx:57]: Not resolved

File: apps/mobile/src/screens/chat/libs/components/chat-box/chat-box.tsx [hunk 23]
```diff
@@ -1,15 +1,44 @@
...
 };
 
-const ChatBox: React.FC<Properties> = ({ children, style }: Properties) => {
-	return <View style={[styles.container, style]}>{children}</View>;
+const ChatBox: React.FC<Properties> = ({
+	author,
+	payload,
+	type,
+}: Properties) => {
+	const isUser = author === "user";
```
[the remark this answers is not in the document]
Comment (fshabanov) [reply apps/mobile/src/screens/chat/libs/components/chat-box/chat-box.tsx:23]: Not resolved

File: apps/mobile/src/screens/chat/libs/components/chat-box/libs/components/components.ts (hunk 2 above)
Comment (fshabanov) [reply apps/mobile/src/screens/chat/libs/components/chat-box/libs/components/components.ts:1]: Not resolved

File: apps/mobile/src/screens/chat/libs/components/chat-buttons/libs/components/checkbox-buttons/checkbox-buttons.tsx (hunk 3 above)
Comment (fshabanov) [reply apps/mobile/src/screens/chat/libs/components/chat-buttons/libs/components/checkbox-buttons/checkbox-buttons.tsx:55]: Not resolved

File: apps/mobile/src/screens/chat/libs/components/chat-buttons/libs/components/generate-task-buttons/generate-task-buttons.tsx (hunk 4 above)
Comment (fshabanov) [reply apps/mobile/src/screens/chat/libs/components/chat-buttons/libs/components/generate-task-buttons/generate-task-buttons.tsx:29]: Not resolved

File: apps/mobile/src/screens/chat/libs/components/chat-buttons/libs/components/generate-task-buttons/generate-task-buttons.tsx [hunk 24]
```diff
@@ -0,0 +1,76 @@
...
+	return (
+		<ChatMessage isUser={false} text={CONFIRMATION_TEXT}>
+			<View style={[globalStyles.gap12, globalStyles.mt12]}>
+				<Button
+					label="Yes, 3 lowest"
+					onPress={handleCreationTasksForThreeLowest}
+				/>
+				<Button
+					appearance="outlined"
+					label="No smth else"
```
Comment (fshabanov) [inline apps/mobile/src/screens/chat/libs/components/chat-buttons/libs/components/generate-task-buttons/generate-task-buttons.tsx:68]:
```suggestion
					label="No, something else"
```

File: apps/mobile/src/screens/chat/libs/components/initial-chat-message/initial-chat-message.tsx [hunk 25]
```diff
@@ -0,0 +1,37 @@
...
+
+const InitialChatMessage: React.FC = () => {
+	const user = useAppSelector((state) => state.auth.user);
+
+	const userName = user?.name ?? "";
+
+	return (
+		<>
+			<ChatMessage
+				isUser={false}
```
Comment (fshabanov) [inline apps/mobile/src/screens/chat/libs/components/initial-chat-message/initial-chat-message.tsx:17]:
```suggestion
```

File: apps/mobile/src/screens/chat/libs/components/initial-chat-message/initial-chat-message.tsx [hunk 26]
```diff
@@ -0,0 +1,37 @@
...
+				style={globalStyles.mb8}
+				text={`Hello ${userName}! I'm so glad you're here and taking steps towards a more balanced life. You've got this!`}
+			/>
+			<ChatMessage isUser={false} text="Here are your Wheel results:">
```
Comment (fshabanov) [inline apps/mobile/src/screens/chat/libs/components/initial-chat-message/initial-chat-message.tsx:21]:
```suggestion
			<ChatMessage text="Here are your Wheel results:">
```

File: apps/mobile/src/screens/chat/libs/components/loader/libs/hooks/use-animated-scale-styles.hook.ts [hunk 27]
```diff
@@ -0,0 +1,12 @@
+import { type SharedValue, useAnimatedStyle } from "react-native-reanimated";
+
+type Property = {
```
Comment (fshabanov) [inline apps/mobile/src/screens/chat/libs/components/loader/libs/hooks/use-animated-scale-styles.hook.ts:3]:
```suggestion
type ReturnValue = {
```

File: apps/mobile/src/screens/chat/libs/components/loader/loader.tsx [hunk 28]
```diff
@@ -0,0 +1,55 @@
...
+			);
+		}
+	}, [scaleDot1, scaleDot2, scaleDot3]);
+
+	const animatedStyle1 = useAnimatedScaleStyle(scaleDot1);
+	const animatedStyle2 = useAnimatedScaleStyle(scaleDot2);
+	const animatedStyle3 = useAnimatedScaleStyle(scaleDot3);
+
+	return (
+		<ChatMessage isUser={false} style={globalStyles.mt8}>
```
Comment (fshabanov) [inline apps/mobile/src/screens/chat/libs/components/loader/loader.tsx:45]:
```suggestion
		<ChatMessage style={globalStyles.mt8}>
```

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/mobile/src/navigations/bottom-tabs-navigator/bottom-tabs-navigator.tsx (hunk 20 above)
[the remark this answers is not in the document]
Comment (KotenkoNata) [reply apps/mobile/src/navigations/bottom-tabs-navigator/bottom-tabs-navigator.tsx:53]: Sorry, maybe I haven't updated the page, and that's why I didn't fix some comments

**Review by anogire — APPROVED** [review]

**Review by revenets — APPROVED** [review]

File: apps/mobile/src/libs/components/components.tsx [hunk 29]
```diff
@@ -32,9 +34,13 @@ export { TaskStatusLabel } from "./task-status-label/task-status-label";
 export { Text } from "./text/text";
 export { ToastMessage } from "./toast-message/toast-message";
 export { TypingTextView } from "./typing-text-view/typing-text-view";
+export { UserAvatar } from "./user-avatar/user-avatar";
 export { Wheel } from "./wheel/wheel";
 export { WheelLoader } from "./wheel-loader/wheel-loader";
 export { default as MaskedView } from "@react-native-masked-view/masked-view";
+export { ChatBox } from "~/screens/chat/libs/components/chat-box/chat-box";
+export { InitialChatMessage } from "~/screens/chat/libs/components/initial-chat-message/initial-chat-message";
+export { FlatList } from "react-native";
```
Comment (fshabanov) [inline apps/mobile/src/libs/components/components.tsx:43]: We already have export from `react-native`, combine

File: apps/mobile/src/screens/chat/libs/components/chat-buttons/libs/components/accept-tasks/accept-tasks.tsx [hunk 30]
```diff
@@ -0,0 +1,82 @@
...
+		dispatch(
+			chatActions.addTextMessage({
+				author: ChatMessageAuthor.USER,
+				text: "Generate new",
+			}),
+		);
+	}, [dispatch, taskSuggestions, threadId]);
+
+	return (
+		<ChatMessage isUser={false} text={FEEDBACK_TEXT}>
```
Comment (fshabanov) [inline apps/mobile/src/screens/chat/libs/components/chat-buttons/libs/components/accept-tasks/accept-tasks.tsx:69]:
```suggestion
		<ChatMessage text={FEEDBACK_TEXT}>
```
Check in other places as well

File: apps/mobile/src/screens/chat/libs/components/chat-buttons/libs/components/generate-task-buttons/generate-task-buttons.tsx [hunk 31]
```diff
@@ -0,0 +1,76 @@
...
+		dispatch(
+			chatActions.addTextMessage({
+				author: ChatMessageAuthor.USER,
+				text: "Yes, 3 lowest",
+			}),
+		);
+	}, [dispatch, threadId, lowestCategories]);
+
+	return (
+		<ChatMessage isUser={false} text={CONFIRMATION_TEXT}>
```
Comment (fshabanov) [inline apps/mobile/src/screens/chat/libs/components/chat-buttons/libs/components/generate-task-buttons/generate-task-buttons.tsx:60]:
```suggestion
		<ChatMessage text={CONFIRMATION_TEXT}>
```

File: apps/mobile/src/screens/chat/libs/components/chat-buttons/libs/components/generate-task-buttons/generate-task-buttons.tsx (hunk 24 above)
Comment (fshabanov) [reply apps/mobile/src/screens/chat/libs/components/chat-buttons/libs/components/generate-task-buttons/generate-task-buttons.tsx:68]: Not fixed

File: apps/mobile/src/screens/chat/libs/components/loader/libs/hooks/use-animated-scale-styles.hook.ts [hunk 32]
```diff
@@ -0,0 +1,12 @@
+import { type SharedValue, useAnimatedStyle } from "react-native-reanimated";
+
+type ReturnValue = {
+	transform: { scale: number }[];
+};
+
+const useAnimatedScaleStyle = (scaleValue: SharedValue<number>): ReturnValue =>
+	useAnimatedStyle(() => ({
+		transform: [{ scale: scaleValue.value }],
+	}));
```
Comment (fshabanov) [inline apps/mobile/src/screens/chat/libs/components/loader/libs/hooks/use-animated-scale-styles.hook.ts:10]: B27

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/mobile/src/screens/chat/libs/components/loader/libs/hooks/use-animated-scale-styles.hook.ts [hunk 33]
```diff
@@ -0,0 +1,14 @@
...
+type ReturnValue = {
+	transform: { scale: number }[];
+};
+
+const useAnimatedScaleStyle = (scaleValue: SharedValue<number>): ReturnValue =>
+	useAnimatedStyle(() => {
+		return {
+			transform: [{ scale: scaleValue.value }],
+		};
+	});
```
Comment (fshabanov) [inline apps/mobile/src/screens/chat/libs/components/loader/libs/hooks/use-animated-scale-styles.hook.ts:12]:
```suggestion
const useAnimatedScaleStyle = (scaleValue: SharedValue<number>): ReturnValue => {
	return useAnimatedStyle(() => {
		return {
			transform: [{ scale: scaleValue.value }],
		};
	}});
```

**Review by fshabanov — APPROVED** [review]

**Pull request merged**